pkg/accelerator/accelDeviceProvider.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accelerator
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+	"github.com/jaypipes/ghw"
+)
+
+type accelDeviceProvider struct {
+	deviceList []types.AccelDevice
+	rFactory   types.ResourceFactory
+}
+
+// NewAccelDeviceProvider DeviceProvider implementation from accelDeviceProvider instance
+func NewAccelDeviceProvider(rf types.ResourceFactory) types.DeviceProvider {
+	return &accelDeviceProvider{
+		rFactory:   rf,
+		deviceList: make([]types.AccelDevice, 0),
+	}
+}
+
+func (ap *accelDeviceProvider) GetDevices() []types.PciDevice {
+	newPciDevices := make([]types.PciDevice, len(ap.deviceList))
+	for i := range ap.deviceList {
```

Comment:
you could just:
```
return copy(newPciDevice, ap.deviceList)
```

Reply:
You cannot use copy with slice of different types.

Reply:
ah missed that 👍